extension/entitystore/serviceprovider.go

```diff
@@ -245,26 +245,28 @@ func (s *serviceprovider) scrapeIAMRole() error {
 	return nil
 }
 func (s *serviceprovider) scrapeImdsServiceNameAndASG() error {
-	tags, err := s.metadataProvider.InstanceTags(context.Background())
+	tagKeys, err := s.metadataProvider.InstanceTags(context.Background())
 	if err != nil {
 		s.logger.Debug("Failed to get service name from instance tags. This is likely because instance tag is not enabled for IMDS but will not affect agent functionality.")
 		return err
 	}
-	// This will check whether the tags contains SERVICE, APPLICATION, APP, in that order.
-	for _, value := range priorityMap {
-		if strings.Contains(tags, value) {
-			serviceName, err := s.metadataProvider.InstanceTagValue(context.Background(), value)
+
+	// This will check whether the tags contains SERVICE, APPLICATION, APP, in that order (case insensitive)
+	lowerTagKeys := toLowerKeyMap(tagKeys)
+	for _, potentialServiceNameKey := range priorityMap {
```

Comment:
nit: I know it's not your code but priorityMap isn't a map, it's a list. Should be `priorityList` or something